src/org/labkey/targetedms/model/RawMetricDataSet.java

```diff
@@ -47,14 +47,61 @@
     String filePath;
     int replicateId;
 
+    String modifiedSequence;
+    String customIonName;
+    String ionFormula;
     Double massMonoisotopic;
     Double massAverage;
     String precursorCharge;
 
     @Nullable
     public String getSeriesLabel()
     {
-        return seriesLabel;
+        if (null != seriesLabel)
+        {
+            return seriesLabel;
+        }
+        else if (null != modifiedSequence)
+        {
+            return modifiedSequence;
```

Comment:
Overall structure looks as I'd expect, but we can't just use modifiedSequence - we need to include the mZ and charge state - please double-check against the SQL that Vagisha proposed in the issue.